app/forms/upload_form.rb

```diff
@@ -15,30 +15,14 @@ class UploadForm
   validates :search_engine, inclusion: { in: SearchResult.search_engines.keys }
   validates :csv_file, presence: true
 
-  validate :valid_content_type, if: -> { csv_file.present? }
-  validate :valid_count, if: -> { csv_file.present? }
+  validate :validate_content_type, if: -> { csv_file.present? }
+  validate :validate_count, if: -> { errors.empty? && csv_file.present? }
```

Comment:
The naming is not ideal as it is repeats `validate`:

```suggestion
  validate :csv_file_content_type, if: -> { csv_file.present? }
  validate :keyword_count, if: -> { errors.empty? && csv_file.present? }
```